pkg/endpoint/bpf.go

```diff
@@ -1120,28 +1126,23 @@ func (e *Endpoint) addPolicyKey(keyToAdd policy.Key, entry policy.MapStateEntry,
 
 // ApplyPolicyMapChanges updates the Endpoint's PolicyMap with the changes
 // that have accumulated for the PolicyMap via various outside events (e.g.,
-// identities added / deleted).
-// 'proxyWaitGroup' may not be nil.
-func (e *Endpoint) ApplyPolicyMapChanges(proxyWaitGroup *completion.WaitGroup) error {
+// identities added / deleted) by triggering the sync-policy-map controller.
+// Returns a channel that is closed when the changes are applied. The channel
+// is closed even if errors occurred. The controller will retry the changes.
+func (e *Endpoint) ApplyPolicyMapChanges() <-chan struct{} {
 	if err := e.lockAlive(); err != nil {
-		return err
+		return nil
```

Comment:
1. reading from a nil channel blocks the reader forever; 
2. there's a window between the calls to `GetEndpoints` and `e.ApplyPolicyMapChanges` below

hence this can block the controller until the context expires, which isn't ideal.

Reply:
Damn it, misremembered the semantics. Will fix to return a closed channel.